modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/utils/dataUtils.ts

```diff
@@ -5,6 +5,86 @@
 
 type LocalUIData = APISchemaUIData | APIApplicationUIData;
 
+interface FieldsDataToProperties {
+	apiSchema: APISchemaItem;
+	objectDefinitions: ObjectDefinition[];
+	schemaProperties: APISchemaPropertyItem[];
+}
+
+export function AddObjectFieldsDataToProperties({
+	apiSchema,
+	objectDefinitions,
+	schemaProperties,
+}: FieldsDataToProperties) {
+	const propertiesTreeViewItems = schemaProperties.map(
+		({description, id, name, objectFieldERC, objectRelationshipNames}) => {
+			const objectRelationshipNamesArray = objectRelationshipNames?.split(
+				','
+			);
+
+			const objectRelationshipName =
+				objectRelationshipNamesArray?.[
+					objectRelationshipNamesArray.length - 1
+				];
+
+			const mainObjectDefinition = objectDefinitions.find(
+				(definition) =>
+					definition.externalReferenceCode ===
+					apiSchema.mainObjectDefinitionERC
+			);
+
+			let objectDefinitionId2: number;
+
+			objectDefinitions.forEach((definition) => {
+				definition.objectRelationships.forEach((relationship) => {
+					{
+						if (relationship.name === objectRelationshipName) {
+							objectDefinitionId2 =
+								relationship.objectDefinitionId2;
+						}
+					}
+				});
+			});
+
+			const relatedDefinition = objectDefinitions.find(
```

Comment:
@cgmonte Please, rename it to `relatedObjectDefinition`. We must avoid refering to object's entities without the 'object' prefix.